modules/openapi-generator/src/main/resources/typescript/api/api.mustache

```diff
@@ -66,7 +66,27 @@ export class {{classname}}RequestFactory extends BaseAPIRequestFactory {
 
         // Query Params
         if ({{paramName}} !== undefined) {
+            {{#isPrimitiveType}}
             requestContext.setQueryParam("{{baseName}}", ObjectSerializer.serialize({{paramName}}, "{{{dataType}}}", "{{dataFormat}}"));
+            {{/isPrimitiveType}}
+            {{^isPrimitiveType}}
+            {{#isExplode}}
+            const serializedParams = ObjectSerializer.serialize({{paramName}}, "{{{dataType}}}", "{{dataFormat}}");
+            {{#isArray}}
+            for (const serializedParam of serializedParams) {
+                requestContext.appendQueryParam("{{baseName}}", serializedParam);
+            }
+            {{/isArray}}
+            {{^isArray}}
+            for (const key of Object.keys(serializedParams)) {
+                requestContext.appendQueryParam(key, serializedParams[key]);
+            }
+            {{/isArray}}
+            {{/isExplode}}
+            {{^isExplode}}
```

Comment:
for unexploded non-primitives we keep the current serialization `name=value1,value2,value3`